docs/rootfs.md

```diff
@@ -0,0 +1,51 @@
+# Getting a rootfs
+
+There are many ways to produce a directory to pass to the `rootfs` config field,
+here are a couple of potential solutions.
+
+## From a container image
+
+OCI images can be turned into tarballs which can be extracted into a rootfs. For
+example:
+
+```sh
+❯❯ mkdir $rootfs_dir && cd $rootfs_dir
+❯❯ cat > Containerfile
+FROM docker.io/library/debian
+RUN apt update
+RUN apt install -y qemu-guest-agent
+
+❯❯ podman build -t deb-qga  # Docker would work exactly the same
+❯❯ podman export -o deb.tar $(podman create deb-qga)
+❯❯ tar xf deb.tar
+❯❯ rm Containerfile deb.tar
+```
+
+## Using mkosi
+
+[`mkosi`](https://github.com/systemd/mkosi) is a more advanced tool for building
+OS images, as well as just producing a rootfs it can build full disk images with
+a bootloader, plus many other features. You'll need to refer to the full
+documentation to really understand `mkosi`, but here's a minimal example. This
+will only work if you host system has `apt`, otherwise you'll need to adapt it
+for your host distro or run it in a container.
+
+`mkosi.conf`:
+
+```ini
+[Output]
+Format=directory
+
+[Distribution]
+Distribution=debian
+Release=testing
```

Comment:
I actually just realised it's dumb to use Debian testing for this. The whole reason I was interested in having a standalone rootfs is to have a way to avoid "works on my machine" situations. But Debian testing is a super unstable target, so it's a strange choice for the example.

Reply:
OK yeah just switching `testing` to `bookworm` seems to work fine. I also tested it on Ubuntu and added a note on a packge you'll need to install. I also discovered an [issue with using mkosi](https://github.com/systemd/mkosi/issues/3265) there, I guess that's too much distro-specific detail to try and explain here though?